### PR TITLE
Customer Edit Screen

### PR DESCRIPTION
**Pull request by Polsander**
Customer edit screen is fully functional.

Please test it, play around, and try to break it. Do different input permutations, leave some blank fields, see what happens.

This will be a really thick PR, so please thoroughly go through the main components and server controller logic. 
I've added a few helpers as well, so feel free to look at them so that you may also be able to use them in your future code edits.

### CODE REVIEW

**Comment by Polsander** [comment]
Refactored this to not be a simple res.send screen anymore.
Implemented the following:

- global state CRUD page rendering
- Multiple reusable components
- Table optimization and adjustments
- Backend edit page submission logic (as well as create, update, delete actions)
- Global state spinner
- Styling

File: microservices/client/src/containers/customers/EditCustomer.tsx
Comment (ferncabrera) [inline microservices/client/src/containers/customers/EditCustomer.tsx:1]: LOL I got a "Warning large diffs are not loaded by default" 😭

File: microservices/client/src/containers/customers/EditCustomer.tsx [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+                <Col className='mrp-50' md={3}>
+                  <Form.Select
+                    as={Col}
+                    value={customerData.vendor}
+                    onChange={(e) => setCustomerData({ ...customerData, vendor: e.target.value })}
+                  >
+                    <option value={''}>Select Available Vendor</option>
+                    {initialVendor && <option value={initialVendor}>{initialVendor}</option>}
+
+                    {!_.isEmpty(vendorList) && _.map(vendorList, (vendor) => {
```
Comment (ferncabrera) [inline microservices/client/src/containers/customers/EditCustomer.tsx:169]: Makes me wanna get familiar with lodash lib.. handy shortcuts

File: microservices/server/src/db/index.ts
Comment (ferncabrera) [inline microservices/server/src/db/index.ts:1]: lovely

**Review by ferncabrera — COMMENTED** [review]
Beautiful code, I have nothing to say but knitpicks:

image
Im sure you already tried but this is not purple like the others, do you think thats right? Even if its on the figma? LMK your thoughts.

image
I think it may make more sense to say "link" rather than connect? Again over Figma

image
And finally we should consider our JioPhone 2 mobile userbase.

Please merge in whenever!

File: microservices/client/src/containers/customers/EditCustomer.tsx (hunk 1 above)
Comment (Polsander) [reply microservices/client/src/containers/customers/EditCustomer.tsx:169]: @ferncabrera Very powerful tool! Gotta look up the documentation a few times, but _.isEmpty and _.get are probably the two most used functions I use today.

File: microservices/server/src/db/index.ts
Comment (Polsander) [reply microservices/server/src/db/index.ts:1]: Thanks :))))

File: microservices/client/src/containers/customers/EditCustomer.tsx
Comment (Polsander) [reply microservices/client/src/containers/customers/EditCustomer.tsx:1]: I cried too :(

**Comment by Polsander** [comment]
@ferncabrera 

Good catch! I didn't notice the select isn't purple by default. Will make a global style change that addresses this.
I can do "Link" instead of "Connect", I agree with you there. 
I am however in favor of keeping the first option as Select available vendor, since only vendors that aren't linked can only be chosen.

Will make these changes and merge after @juanitosol gets a chance to review.

As for jioPhone 2, which browser is even showing you that option? Chrome for me at least does not LOL so funny.

**Pull request merged**